### PR TITLE
feat(tailwind-preset): enhanced uiVariants (group/peer/parent, prefix handling)

### PR DESCRIPTION
**Pull request by Dugyu**
## Summary by CodeRabbit

* **New Features**
  * Added UI variant modifiers: group-*, peer-*, and parent-* (ancestor, sibling, direct-parent); group/peer support optional named modifiers.

* **Bug Fixes**
  * Prefix handling: ui-* state markers remain unprefixed; group/peer selectors honor the configured Tailwind prefix.

* **Docs**
  * Expanded usage examples, quick reference, and guidance for self/parent/group/peer scopes; README samples updated with typed config examples.

* **Tests**
  * Expanded coverage for variants, modifiers, prefix mappings, and edge cases.

* **Chores**
  * Bumped Tailwind CSS peer dependency to ^3.4.0; minor version release.

* **Deprecations**
  * Removed slash-based modifiers for self variants; slash syntax retained for scoped markers.

## Examples (quick overview)

**Self**

```tsx
<Button className='ui-open:bg-blue-500' />;
```

```css
.ui-open\:bg-blue-500.ui-open {
  background-color: #3b82f6;
}
```

**Parent**

```tsx
<Parent className='ui-open'>
  <Child className='parent-ui-open:bg-green-500' />
</Parent>;
```

```css
.ui-open > .parent-ui-open\:bg-green-500 {
  background-color: #10b981;
}
```

**Group (Ancestor)**

```tsx
<Ancestor className='group ui-open'>
  <Parent>
    <Item className='group-ui-open:opacity-50' />
  </Parent>
</Ancestor>;
```

```css
.group.ui-open .group-ui-open\:opacity-50 {
  opacity: 0.5;
}
```

With example project prefix `tw-`:

```tsx
<Ancestor className='tw-group ui-open'>
  <Parent>
    <Item className='group-ui-open:tw-opacity-50' />
  </Parent>
</Ancestor>;
```

```css
.tw-group.ui-open .group-ui-open\:tw-opacity-50 {
  opacity: 0.5;
}
```

**Peer (Sibling)**

```tsx
<Input className="peer ui-checked" />
<Label className="peer-ui-checked:text-rose-600" />
```

```css
.peer.ui-checked ~ .peer-ui-checked\:text-rose-600 {
  color: #e11d48;
}
```

With example project prefix `tw-`:

```tsx
<Input className="tw-peer ui-checked" />
<Label className="peer-ui-checked:tw-text-rose-600" />
```

```css
.tw-peer.ui-checked ~ .peer-ui-checked\:tw-text-rose-600 {
  color: #e11d48;
}
```
## Checklist

- [x] Tests updated (or not required).
- [x] Documentation updated (or not required).
- [x] Changeset added, and when a BREAKING CHANGE occurs, it needs to be clearly marked (or not required).

### CODE REVIEW

File: packages/third-party/tailwind-preset/docs/plugins/lynx-ui/uiVariants.md [hunk 1]
```diff
@@ -105,7 +105,198 @@ These variants enable component-aware styling by aligning Tailwind utilities wit
 To make these variants effective, your component needs to append the corresponding `ui-*` class dynamically based on its internal state or configuration. For example:
 
 ```tsx
-<Popover className={isOpen ? 'ui-open container' : 'container'}>
-  <div className='ui-open:bg-blue-500 ui-side-left:border-l' />
+<Popover className={isOpen ? 'container ui-open' : 'container'}>
+  <view className='ui-open:bg-blue-500 ui-side-left:border-l' />
```
Comment (upupming) [inline packages/third-party/tailwind-preset/docs/plugins/lynx-ui/uiVariants.md]: `ui-open:bg-blue-500` generates:

```css
.ui-open\:bg-blue-500.ui-open {
  background-color: #3b82f6;
}
```

It seems `.ui-open` should be added to `view` instead of `Popover` to make this example work.

File: packages/third-party/tailwind-preset/docs/plugins/lynx-ui/uiVariants.md [hunk 2]
```diff
@@ -105,7 +105,198 @@ These variants enable component-aware styling by aligning Tailwind utilities wit
...
+
+The `parent-*` modifier fills the gap between self and group scopes.
+It lets a child react to its **immediate container's** state without requiring any extra marker class:
+
+- **Clear mental model** — "my direct parent controls me."
+- **Safer matching** — avoids accidental wide matches that ancestor‐based `group-*` could introduce.
+- **Efficient** — one level lookup is cheaper than scanning arbitrary ancestor chains.
+- **Prefix simplicity** — no special prefixing rules are needed beyond the parent's `ui-*` state.
+
+> Note: Tailwind itself does not provide a `parent-*` modifier.
```
Comment (upupming) [inline packages/third-party/tailwind-preset/docs/plugins/lynx-ui/uiVariants.md:173]: I have two concerns here:

1. > "my direct parent controls me."

In LynxUI, `className` is passed to a custom component, so it seems to be "my direct parent" by it finally turn to not be:

```jsx
const Comp = ({ className, children }) => {
  <view  className={className}>
    <view>
      {children}
     </view>
  </view>
}

<Comp className="parent">
  <view className="child"/>
</Comp>

// .parent > .child does not work here
```

2. Since it is not supported by tailwindcss by default, adding it maybe a very specific need. Can we let user handle this instead of make it a out-of-box support.

**Review by colinaaa — DISMISSED** [review]

File: packages/third-party/tailwind-preset/docs/plugins/lynx-ui/uiVariants.md [hunk 3]
```diff
@@ -105,7 +105,198 @@ These variants enable component-aware styling by aligning Tailwind utilities wit
...
+
+```tsx
+// Prefix ON markers and utilities:
+<view className="tw-group ui-selected">
+  <view className="group-ui-selected:tw-bg-blue-500" />
+</view>
+
+// Prefix OFF on states:
+<view className="ui-active">
+  <view className="ui-active:tw-text-red-500" />
```
Comment (upupming) [inline packages/third-party/tailwind-preset/docs/plugins/lynx-ui/uiVariants.md]: Shall we use `parent:` of `group:` here?

Comment (Dugyu) [reply packages/third-party/tailwind-preset/docs/plugins/lynx-ui/uiVariants.md]: Yeah, you're right:)

File: packages/third-party/tailwind-preset/docs/plugins/lynx-ui/uiVariants.md (hunk 2 above)
Comment (Dugyu) [reply packages/third-party/tailwind-preset/docs/plugins/lynx-ui/uiVariants.md:173]: Great example! It clearly shows why `.parent > .child` doesn't always hold when className is passed down deeply in UI components. Thanks for pointing that out:)
That said, we still think `parent-*` is helpful in Lynx because:
- Performance: the direct child selector is cheaper than the descendant selector (sensitive in Lynx's runtime, cheap enough in browsers).
- DX: no need to add an extra `.group` marker.
- Consistency: saves users from re-implementing the same variant logic.

So `parent-*` is a good complement to `group-*`/`peer-*`, even if it is not officially implemented in TailwindCSS.

**Review by upupming — APPROVED** [review]
